File: codegen/results/v001/qwen2.5-coder_7b/20260409_103237/typescript/007_beatles_interview/solution.ts
import * as fs from 'fs';
import * as path from 'path';

const inputFilePath = path.join(__dirname, 'input', 'input.csv');
const expectedFormatPath = path.join(__dirname, 'input', 'expected_format.json');

// Read the expected format to understand the output structure
const expectedFormat = JSON.parse(fs.readFileSync(expectedFormatPath, 'utf-8'));

// Function to calculate age as of July 1, 2025
function calculateAge(birthDate: string): number {
    const birthYear = parseInt(birthDate.split('-')[0], 10);
    return 2025 - birthYear;
}

// Read the input CSV file
const inputContent = fs.readFileSync(inputFilePath, 'utf-8');
const lines = inputContent.trim().split('\n');

// Parse the headers and map them to expected keys
const headers = lines[0].split(',');

// Function to transform each row of data
function transformRow(row: string[]): any {
    const values = row.map((value, index) => {
        if (headers[index] === 'dob') {
            return calculateAge(value);
        } else if (expectedFormat[headers[index]]) {
            return expectedFormat[headers[index]](value);
        }
        return value;
    });
    return Object.fromEntries(headers.map((header, index) => [expectedFormat[header] || header, values[index]]));
}

// Transform the data
const transformedData = lines.slice(1).map(transformRow);

// Output the JSON array
console.log(JSON.stringify(transformedData, null, 2));

File: codegen/results/v001/qwen2.5-coder_7b/20260409_103237/python/007_beatles_interview/solution.py
import csv
from datetime import datetime

# Define the input and expected files
input_file = 'input/input.csv'
expected_format_file = 'input/expected_format.json'

# Read the expected format to understand the structure of the output
with open(expected_format_file, 'r') as f:
    expected_output = f.read()

# Parse the expected output to get the keys we need in our JSON array
import json
expected_keys = list(json.loads(expected_output)[0].keys())

# Initialize an empty list to store the results
results = []

# Read the input CSV file
with open(input_file, 'r') as csvfile:
    csvreader = csv.DictReader(csvfile)
    
    for row in csvreader:
        # Calculate age as of July 1, 2025
        dob = datetime.strptime(row['dob'], '%Y-%m-%d')
        target_date = datetime(2025, 7, 1)
        age = target_date.year - dob.year - ((target_date.month, target_date.day) < (dob.month, dob.day))
        
        # Create a dictionary with the required keys and values
        result_row = {
            'name': row['name'],
            'gender': row['gender'],
            'age': age,
            'email': row['email']
        }
        
        # Append the result to the list
        results.append(result_row)

# Convert the list of dictionaries to a JSON array and print it
import json
print(json.dumps(results, indent=2))